src/api.cpp

```diff
@@ -5476,13 +5483,20 @@ cl_program CLVK_API_CALL clCreateProgramWithIL(cl_context context,
 }
 
 cl_int CLVK_API_CALL
-clSetProgramSpecializationConstant(cl_program program, cl_uint spec_id,
+clSetProgramSpecializationConstant(cl_program prog, cl_uint spec_id,
                                    size_t spec_size, const void* spec_value) {
-    TRACE_FUNCTION("program", (uintptr_t)program, "spec_id", spec_id,
-                   "spec_size", spec_size);
+    TRACE_FUNCTION("program", (uintptr_t)prog, "spec_id", spec_id, "spec_size",
+                   spec_size);
     LOG_API_CALL("program = %p, spec_id = %u, spec_size = %zu, spec_value = %p",
-                 program, spec_id, spec_size, spec_value);
-    return CL_INVALID_OPERATION;
+                 prog, spec_id, spec_size, spec_value);
+
+    auto program = icd_downcast(prog);
+
+    if (!is_valid_program(program)) {
+        return CL_INVALID_PROGRAM;
+    }
+
```

Comment:
A few more errors need to be handled here. Please handle them or add TODOs:

• CL_INVALID_OPERATION if no devices associated with program support intermediate language programs.
• CL_COMPILER_NOT_AVAILABLE if program is created with clCreateProgramWithIL and a compiler is not available, i.e. CL_DEVICE_COMPILER_AVAILABLE specified in the Device Queries table is set to CL_FALSE.
• CL_INVALID_VALUE if spec_value is NULL.